app/controllers/api/json/visualizations_controller.rb

```
@@ -71,7 +71,7 @@ def create
                 Visualization::Member.new(vis_data.merge(name: name_candidate, user_id:  current_user_id))
               end
 
-        vis.privacy = vis.default_privacy(current_user)
+        vis.ensure_valid_privacy
```

The privacy and user permissions are already validated at the model. I wouldn't go so far as auto-correcting the privacy in the validation, better to throw an error, so I think calling this from here makes sense.

I can put the default_privacy part into the visualization `initialize`, but I'd keep the `ensure_valid_privacy` call here. Auto-correcting privacy in the model seems wrong, better to throw a validation error.